kmath-ast/src/jvmMain/kotlin/scientifik/kmath/asm/AsmExpressions.kt

```diff
@@ -0,0 +1,247 @@
+package scientifik.kmath.asm
+
+import scientifik.kmath.expressions.Expression
+import scientifik.kmath.expressions.ExpressionAlgebra
+import scientifik.kmath.operations.*
+import kotlin.reflect.full.memberFunctions
+import kotlin.reflect.jvm.jvmName
+
+interface AsmExpression<T> {
+    fun tryEvaluate(): T? = null
+    fun invoke(gen: AsmGenerationContext<T>)
```

Comment:
does it make sense to do invoke without `operator`? Those method need to be documented as well

Reply:
OK

Reply:
It seems like `compile` is a better name.

Reply:
I'll make invoke internal

Reply:
... as well as tryEvalute